lasy/profiles/speckle_profile.py

```diff
@@ -0,0 +1,519 @@
+            -0.5 * (self.n_beamlets[0] - 1),
+            0.5 * (self.n_beamlets[0] - 1),
+            num=self.n_beamlets[0],
+        )
+        self.y_lens_list = np.linspace(
+            -0.5 * (self.n_beamlets[1] - 1),
+            0.5 * (self.n_beamlets[1] - 1),
+            num=self.n_beamlets[1],
+        )
+        self.Y_lens_matrix, self.X_lens_matrix = np.meshgrid(
+            self.y_lens_list, self.x_lens_list
+        )
+        self.Y_lens_index_matrix, self.X_lens_index_matrix = np.meshgrid(
+            np.arange(self.n_beamlets[1], dtype=float),
+            np.arange(self.n_beamlets[0], dtype=float),
+        )
+        self.set_phase_plate_phase_modulation()
+
+        if "SSD" in self.temporal_smoothing_type.upper():
+            # ======================== SSD parameters ========================= #
+            # Only support single FM for now
+            # the amplitude of phase along each direction
+            self.ssd_phase_modulation_amplitude = ssd_phase_modulation_amplitude
+            # number of color cycles
+            self.ssd_number_color_cycles = ssd_number_color_cycles
+            # bandwidth distributed with respect to the two transverse direction
+            self.ssd_transverse_bandwidth_distribution = (
+                ssd_transverse_bandwidth_distribution
+            )
+            ssd_normalization = np.sqrt(
+                self.ssd_transverse_bandwidth_distribution[0] ** 2
+                + self.ssd_transverse_bandwidth_distribution[1] ** 2
+            )
+            ssd_frac = [
+                self.ssd_transverse_bandwidth_distribution[0] / ssd_normalization,
+                self.ssd_transverse_bandwidth_distribution[1] / ssd_normalization,
+            ]
+            self.ssd_phase_modulation_frequency = [
+                self.laser_bandwidth * sf * 0.5 / pma
+                for sf, pma in zip(ssd_frac, self.ssd_phase_modulation_amplitude)
+            ]
+            self.ssd_time_delay = (
+                (
+                    self.ssd_number_color_cycles[0]
+                    / self.ssd_phase_modulation_frequency[0]
+                    if self.ssd_phase_modulation_frequency[0] > 0
+                    else 0
+                ),
+                (
+                    self.ssd_number_color_cycles[1]
+                    / self.ssd_phase_modulation_frequency[1]
+                    if self.ssd_phase_modulation_frequency[1] > 0
+                    else 0
+                ),
+            )
+
+        # ================== Sanity checks on user inputs ===================== #
+        assert (
+            temporal_smoothing_type.upper() in SpeckleProfile.supported_smoothing
+        ), "Only support one of the following: " + ", ".join(
+            SpeckleProfile.supported_smoothing
+        )
+        assert relative_laser_bandwidth > 0, "laser_bandwidth must be greater than 0"
+        for q in (n_beamlets,):
+            assert np.size(q) == 2, "has to be a size 2 array"
```

Comment:
Isn't this equivalent to:
```suggestion
        assert np.size(n_beamlets) == 2, "has to be a size 2 array"
```